libraries/ms-drivers/inc/drv120_relay.h

```diff
@@ -1,17 +1,28 @@
 #pragma once
 
 // Driver for the DRV120 relay
-// Requires GPIO to be initialized
+// Requires GPIO and Gpio_it to be initialized
 
 #include <stdbool.h>
+#include <stdlib.h>
 
 #include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
 #include "status.h"
 
+typedef struct Drv120RelaySettings {
+  GpioAddress *pin;        // Pin initialized as output for relay
+  GpioAddress *status;     // Pin set to trigger interrupt on error
+  GpioItCallback handler;  // Error handler
```

Comment:
Reusing `GpioItCallback` here means that every callback needs a `const GpioAddress *` parameter that they shouldn't have to care about. To make this nicer, I'd suggest `typedef`ing a new `Drv120ErrorCallback` function pointer type that just takes a `void *context`. Then you can just save that callback in a static variable alongside `s_relay_pin` and call it in a static gpio interrupt callback function.